CMakeLists.txt

```diff
@@ -128,7 +135,7 @@ else ()
    endif ()
    set (BSON_EXTRA_ALIGN 1)
    CHECK_SYMBOL_EXISTS(SYS_gettid sys/syscall.h BSON_HAVE_SYSCALL_TID)
-   if (NOT BSON_HAVE_SYSCALL_TID OR APPLE)
+   if (NOT BSON_HAVE_SYSCALL_TID OR APPLE OR ANDROID)
```

Comment:
Curious - was BSON_HAVE_SYSCALL_TID set to 1 on Android when it shouldn't be? You can debug with:
```
message(STATUS "BSON_HAVE_SYSCALL_TID = ${BSON_HAVE_SYSCALL_TID}")
```
I see that my colleague Hannes added the "OR APPLE" predicate with the commit comment, "build libbson on tvOS / watchOS". But it seems to me that CHECK_SYMBOL_EXISTS should do all the necessary checks itself without additional checks on this line?

Reply:
Fixed.
But I think there is a problem with Android NDK r16 and `BSON_HAVE_SYSCALL_TID`.

I'll assess this properly and do another PR for r16 if that's ok.